Pull request from dev to main

makaroveo opened the pull request.

makaroveo committed:
Основные разделы, react router

Добавил основные разделы, основную структуру, добавил многостраничность через React Router

makaroveo committed:
Пересборка идеи

Пришла идея сделать свое портфолио, как репетитора математики, так как это текущая проф деятельность.
Сделаны структуры страниц Главная, О занятиях, Контакты.
Начало разработки Header

makaroveo committed:
Header

Прототип header, начало создания aboutMe

makaroveo committed:
Сделал раздел «Обо мне»

Доделал раздел обо мне, возможно, позже, будут доработаны media запросы

makaroveo committed:
Отредактирован раздел услуги

Сделан раздел услуги, добавлены медиа запросы, изменение стиля при наведении

makaroveo committed:
Раздел отзывов

Сделал каруcель отзывов, медиа запросы, использовал библиотеку React slick

makaroveo committed:
Footer

Создание футера

makaroveo committed:
Добавлен раздел О формате на страницу «О занятих»

Сделал раздел «О формате», реализовал плавное появление блоков с преимуществами

makaroveo committed:
Раздел инструменты

Здесь довольно долго решал вопрос с медиа запросами, они проработоны довольно детально.

Также реализована задача «активного» сервиса

makaroveo committed:
Контакты

Раздел контакты реализован

makaroveo committed:
Блок контаков

Реализовал блок контактов, указал минимальлную высоту для более красивого вида

makaroveo committed:
dependencies devdependencies

dependencies devdependencies изменил, очистил readme

makaroveo committed:
Добавил лого, очистил Readme

makaroveo committed:
Update README.md

makaroveo committed:
Исправил index.html

Исправил index.html, чтобы работали media на устройствах

makaroveo committed:
Merge branch 'dev2' of https://github.com/semonmakarov/react-portfolio into dev2

makaroveo committed:
Merge pull request From fork dev2 to dev

From fork dev2 to dev

stepancar commented on src/App.css:
```diff
@@ -0,0 +1,95 @@
+    outline: none;
+}
+
+img {
```
Судя по логике, которую я тут вижу - этот файл стилей ответственен за общие стили приложения. 

.section-title , .section  уже не относятся к общим стилям.
Давайте наведем порядок

semonmakarov replied:
Исправил, выделил специфические стили в отдельный файл

stepancar commented on src/App.css:
```diff
@@ -0,0 +1,95 @@
+@media screen and (max-width: 768px) {
+}
+
+@media screen and (max-width: 576px) {
+    .section-title {
+        font-size: 28px;
+    }
+}
```
Этот стиль специфичен для компонента который описывает этот элемент

semonmakarov replied:
Исправлено

stepancar commented on src/components/aboutMe/AboutMe.jsx:
```diff
@@ -0,0 +1,30 @@
+        <div className="section aboutMe-section">
+            <div className="section-title">Обо мне</div>
+            <div className="container aboutMe-container">
+                <img src="aboutMe.jpg" className="aboutMe-img" />
```
если вы подключите линте для accessability вы заметите что не хватает некоторых аттрибутов

https://www.npmjs.com/package/eslint-plugin-jsx-a11y

И вам на это укажет автоматика, а ревьюер не будет тратить на это время

semonmakarov replied:
Исправил

stepancar commented on src/components/aboutMe/AboutMe.jsx:
```diff
@@ -0,0 +1,30 @@
+            <div className="section-title">Обо мне</div>
+            <div className="container aboutMe-container">
+                <img src="aboutMe.jpg" className="aboutMe-img" />
+                <div className="aboutMe-information">
+                    <div className="aboutMe-title-container">
```
Обратите внимание что при таком подходе вы можете столкнуться с тем что вам нужно поддерживать уникальность имени классов или же заботиться о повышении специфичности селекторов. Вместе с react-create-app в комплекте идет https://github.com/css-modules/css-modules
Попробуйте использовать их

semonmakarov replied:
Благодарю! Использовал, изменил большую часть проекта

stepancar commented on src/components/contact/Contact.jsx:
```diff
@@ -0,0 +1,28 @@
+            <div className="section-title">Контакты</div>
+            <div className="container contact-container">
+                <div className="contact-info">
+                    <h3>Свяжитесь со мной:</h3>
+                    <p>Телефон: +7 (933) 200-1407</p>
```
Давайте сделаем так чтобы при нажатии на телефон открывался стандартный диалог для звонка

semonmakarov replied:
Сделано

stepancar commented on src/components/contact/Contact.jsx:
```diff
@@ -0,0 +1,28 @@
+                <div className="contact-info">
+                    <h3>Свяжитесь со мной:</h3>
+                    <p>Телефон: +7 (933) 200-1407</p>
+                    <p>Email: semon.makarov.2003@mail.ru</p>
```
Давайте сделаем так, чтобы при нажатии открывался стандартный почтовый клиент

semonmakarov replied:
Сделано

stepancar commented on package.json:
```diff
@@ -0,0 +1,44 @@
+    "react-slick": "^0.30.2",
+    "slick-carousel": "^1.8.1"
+  },
+  "scripts": {
```
Не хватает автоматизации проверки кода на соответствие гайдлайна. Добавьте eslint/ csslint, интегрируйте с github actions

stepancar reviewed (changes requested):
Оставил комментарии

makaroveo committed:
Структуризация стилей в App.css

makaroveo committed:
Разделение App.css на общие стили и специфичные

makaroveo committed:
eslint

makaroveo committed:
Контакты/ e-mail

А также начало работы с css modules

makaroveo committed:
Хочу попробовать реализовать через webpack.config.dev.js

makaroveo committed:
CSS modules всего проекта

Пересмотрел все файлы проекта, изменил название классов, используя camelCase нотацию для более удобного использования классов в js

makaroveo committed:
eslint

makaroveo committed:
Create .eslintrc.yml

makaroveo committed:
Merge pull request #17 from semonmakarov/dev2

Dev2

semonmakarov commented:
Осталась реализация GitHub Actions

makaroveo committed:
eslint

makaroveo committed:
Merge pull request #18 from semonmakarov/dev2

eslint

makaroveo committed:
Merge pull request #19 from stepancar-web-programming/main

gith act

makaroveo committed:
@microsoft/eslint-formatter-sarif

makaroveo committed:
Merge pull request #20 from semonmakarov/dev2

@microsoft/eslint-formatter-sarif

makaroveo committed:
Update eslint.yml

makaroveo committed:
Update eslint.yml

makaroveo committed:
Update3 eslint.yml

stepancar commented on src/specificStyles.css:
```diff
@@ -0,0 +1,39 @@
+    font-weight: 800;
+    margin-bottom: 2rem;
+    text-transform: uppercase;
+    /* font-family: Poppins, sans-serif; */
```
Давайте наведем порядок, удалим то, что не нужно

semonmakarov replied:
Сделано

stepancar commented on package.json:
```diff
@@ -0,0 +1,146 @@
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^27.4.3",
+    "jest-resolve": "^27.4.2",
+    "jest-watch-typeahead": "^1.0.0",
```
У нас с вами выше уже кажется была дискуссия про dependencies/devDependencies. Давайте еще раз разберемся в этом вопросе

semonmakarov replied:
То, что было добавлено в dependencies при установке reject нужно вручную перенести в dependencies/devDependencies?

stepancar commented on scripts/start.js:
```diff
@@ -0,0 +1,154 @@
+'use strict';
```
Можете пояснить по какой причине вам пришлось сделать eject?

semonmakarov replied:
Была идея реализовать css modules через изменение файла webpack.config.dev.js, поэтому сделал eject
https://habr.com/ru/articles/335244/
https://stackoverflow.com/questions/60077172/running-npm-run-eject-to-use-css-modules
Но потом было решено реализовать «по-обычному»

semonmakarov replied:
Если требуется, я могу перезалить проект заново, добавив файлы в новое приложение

stepancar commented on .github/workflows/eslint.yml:
```diff
@@ -1,30 +1,36 @@
-      - name: Install Dependencies
-        run: npm ci
+      - name: Run ESLint and create sarif file
+        run: npx eslint . --config .eslintrc.js --ext .js,.jsx,.ts,.tsx --format @microsoft/eslint-formatter-sarif --output-file eslint-results.sarif
```
обратите внимание на то, что кофигурация локально и на ci отличается.

Самый простой способ добиться консистентности - избегать передачи кастомных параметров в ci скриптах.

Тут должно быть что-то очень простое

тип npm run lint

stepancar replied:
npm скрипты как раз и берут на себя инкапсуляцию всей логики.

Это похоже на паттерн в других языках программирования когда скрипты выносят в makefile

makaroveo committed:
Удаление «мусора» в specificStyles.css

makaroveo committed:
editorcongif + testing.yml

makaroveo committed:
Update CSS module import path

makaroveo committed:
Перезалив в нужную папку yaml

makaroveo committed:
add test

makaroveo committed:
Merge branch 'dev' into dev4

makaroveo committed:
Merge pull request from fork dev4 to dev

pull from fork dev4 to dev

stepancar reviewed (approved):
Как только будут починены проверки на CI и конфликты - пожалуйста, смержите ПР

semonmakarov commented:
> Как только будут починены проверки на CI и конфликты - пожалуйста, смержите ПР
Степан Денисович, такой вопрос.
Было решено создать новую ветку dev2 (откуда вы можете увидеть pull request в  main), где я откатил проекта до установки eject, так с ним возникало большое количество ошибок при разработке GitHub Actions.
Как лучше поступить? Очистить текущую ветку и сделать merge с исправленным проектом? Или залить можно проект в main в dev2, чтобы текущий pull остался?

stepancar commented:
@semonmakarov , по хорошему, желательно сделать все чтобы вмержить ПР, в котором мы вели дискуссию, тогда история сохранится. Но в данном проекте это не принципиально

stepancar commented:
@semonmakarov , пожалуйста закройте все не актуальные Prs
